### PR TITLE
Read from stdin

### PR DESCRIPTION
**Pull request by RyanZim**
Attn: @michael-ciniawsky

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -76,19 +78,26 @@ if (!output && !dir && !argv.replace) {
...
-    let files = arr[0]
+getConfig({}, argv.config)
+  .then(() => {
+    spinner.succeed()
 
-    if (!files || !files.length) throw new Error('You must pass a list of files to parse')
+    if (input && input.length) return globber(input)
+    // Read from stdin:
+    else {
+      if (!argv.output) throw new Error('Must specify --output when reading from stdin')
```
Comment (michael-ciniawsky) [inline index.js]: Why is this line explicitly needed when reading from `process.stdin`? Is L74 not throwing an error anyways if no output is specified? What about letting it write to `process.stdout` instead of throwing an error? That should also enable piping `|` then.

```js
if ( !dir || !output || argv.replace ) chalk.bold.yellow(`
  No Output specified, it's recommended to pass either --output, --dir, or --replace to PostCSS CLI. 
  Result will be written to process.stdout (Console)
`)
```


```bash
postcss index.sss -p 'sugarss' | cleancss
```

File: index.js [hunk 2]
```diff
@@ -76,19 +78,26 @@ if (!output && !dir && !argv.replace) {
...
 console.warn(chalk.bold.red(logo))
 
-spinner.text = `Loading Config`
+spinner.text = 'Loading Config'
 spinner.start()
-Promise.all([ globber(input), getConfig({}, argv.config) ])
-  .then((arr) => {
-    // Until parameter destructuring is supported
-    let files = arr[0]
+getConfig({}, argv.config)
```
Comment (michael-ciniawsky) [inline index.js]: Maybe it is better/necessary to move config loading into processCSS to expose the current file to `postcss.config.js`and load the config **per file**

```js
module.exports = ({ options, file, env }) => ({ // { options, file, env } === ctx && options = argv.ctx
  parser: file.extname === '.sss' ? 'sugarss' : false
  map: options.map || false
  plugins: {
     'postcss-modules':  options.modules ? {} : false
     'cssnano':  env === 'production' ? options.minify : false
  }
})
```

File: index.js [hunk 3]
```diff
@@ -76,19 +78,26 @@ if (!output && !dir && !argv.replace) {
...
+  .then(() => {
+    spinner.succeed()
 
-    if (!files || !files.length) throw new Error('You must pass a list of files to parse')
+    if (input && input.length) return globber(input)
+    // Read from stdin:
+    else {
```
Comment (michael-ciniawsky) [inline index.js]:
```diff
- else
```

File: index.js [hunk 4]
```diff
@@ -149,13 +156,26 @@ function processCSS (file, watcher) {
...
 
+function processFiles (files, watcher) {
+  if (typeof files === 'string') files = [files]
+  return Promise.all(files.map(file => {
+    // Read from stdin
+    if (file === '-') {
+      return getStdin()
+      .then(css => processCSS(css, undefined, watcher))
+    }
+    return fs.readFile(file)
```
Comment (michael-ciniawsky) [inline index.js]: `read-cache`

File: index.js (hunk 1 above)
Comment (RyanZim) [inline index.js]: I'm thinking about adding process.stdout support in a later PR.

The main reason for this line is to throw an error if `--dir` or `--replace` is passed, since we don't have a filename to write.

Comment (RyanZim) [inline index.js]: I'm thinking about adding process.stdout support in a later PR.

The main reason for this line is to throw an error if --dir or --replace is passed, since we don't have a filename to write.

File: index.js (hunk 4 above)
Comment (RyanZim) [inline index.js]: That should be a separate PR IMO, will do though.

File: index.js (hunk 1 above)
Comment (RyanZim) [inline index.js]: This is mainly to throw an error if `-d` or `-r` is passed. I'm thinking about enabling stdout support, just haven't done yet since v2 didn't support stdout (v2 feature parity is first priority).

File: index.js (hunk 2 above)
Comment (RyanZim) [inline index.js]: Let's do that later

File: index.js (hunk 4 above)
Comment (michael-ciniawsky) [inline index.js]: Yep lets do it in the finalise step

File: index.js (hunk 2 above)
Comment (michael-ciniawsky) [inline index.js]: Yep thats related to #72

**Comment by michael-ciniawsky** [comment]
With process.stdin && process.stdout support we should check options for refactoring to `streams` where possible it will make the code easier and cleaner https://github.com/postcss/postcss-cli/issues/51#issuecomment-270295282 (just rough prototype)

**Comment by RyanZim** [comment]
@michael-ciniawsky To be honest, I don't see that happening for v9.0.0; I haven't even investigated whether it's a good idea or not. Let's make sure we've got a working product and ship v9.0.0, then continuously improve on that.

**Comment by michael-ciniawsky** [comment]
`v9` ? 😛 

Hmm...Well I know what you mean, but... I will try out otherwise keep it for now. IMHO

```bash
lessc | postcss -u autoprefixer | cleancss
```

should work (pipeable (in/out))

**Comment by RyanZim** [comment]
Yeah, that I want in v9, but streams internally...No.
